From 1f11b1356115a823921fcc6cf437d55a5d05fc7c Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Wed, 29 Sep 2021 22:44:16 -0400
Subject: [PATCH 05/10] Fix test caused by shell optimization

Both bash and zsh seems to exec the last command when possible
causing the command name for the process to change.

Add a `; true` after the readlink to make sure this doesn't happen.
---
 test/test-99999999.sh | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/test/test-99999999.sh b/test/test-99999999.sh
index f0a8c84..092000e 100644
--- a/test/test-99999999.sh
+++ b/test/test-99999999.sh
@@ -39,8 +39,11 @@ TEST=$(${PROOT} -b /proc -r ${ROOTFS} readlink /proc/self/fd/0 | grep -E "^/proc
 test -z $TEST
 
 if [ ! -z $$ ]; then
-TEST=$(readlink -f /proc/$$/exe)
-${PROOT} sh -c 'true; readlink /proc/$$/exe' | grep ${TEST}
+    TEST=$(readlink -f /proc/$$/exe)
+    # The `true` after the readlink makes sure that
+    # the shell doesn't `exec` the `readlink` command and causes the process exe
+    # to be readlink instead
+    ${PROOT} sh -c 'readlink /proc/$$/exe; true' | grep ${TEST}
 fi
 
 MD5=$(md5sum $(which md5sum) | cut -f 1 -d ' ')
-- 
2.33.0

